src/binder_voicecall.c

```diff
@@ -609,38 +688,54 @@ binder_voicecall_clcc_poll_cb(
     if (status == RADIO_TX_STATUS_OK) {
         if (error == RADIO_ERROR_NONE) {
             GBinderReader reader;
-            gsize i, count = 0;
+            gsize i;
 
             /* getCurrentCallsResponse(RadioResponseInfo, vec<Call> calls); */
             gbinder_reader_copy(&reader, args);
-            if (resp == RADIO_RESP_GET_CURRENT_CALLS) {
-                const RadioCall* calls =
-                    gbinder_reader_read_hidl_type_vec(&reader,
-                        RadioCall, &count);
 
-                if (calls) {
-                    /* Build sorted list */
-                    for (i = 0; i < count; i++) {
-                        list = g_slist_insert_sorted(list,
-                            binder_voicecall_info_new(calls + i),
-                            binder_voicecall_info_compare);
+            if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+                gsize count = 0;
+                if (resp == RADIO_RESP_GET_CURRENT_CALLS) {
+                    const RadioCall* calls =
+                        gbinder_reader_read_hidl_type_vec(&reader,
+                            RadioCall, &count);
+
+                    if (calls) {
+                        /* Build sorted list */
+                        for (i = 0; i < count; i++) {
+                            list = g_slist_insert_sorted(list,
+                                binder_voicecall_info_new(calls + i),
+                                binder_voicecall_info_compare);
+                        }
+                    }
+                } else if (resp == RADIO_RESP_GET_CURRENT_CALLS_1_2) {
+                    const RadioCall_1_2* calls =
+                        gbinder_reader_read_hidl_type_vec(&reader,
+                            RadioCall_1_2, &count);
+
+                    if (calls) {
+                        /* Build sorted list */
+                        for (i = 0; i < count; i++) {
+                            list = g_slist_insert_sorted(list,
```

Comment:
Twice the same functionality with only one changing variable. Yes. Lets call it a function?